app/scripts/lib/createDupeReqFilterMiddleware.test.ts

```diff
@@ -0,0 +1,111 @@
+import { jsonrpc2 } from '@metamask/utils';
+import createDupeReqFilterMiddleware, {
+  THREE_MINUTES,
+} from './createDupeReqFilterMiddleware';
+
+describe('createDupeReqFilterMiddleware', () => {
+  const getMockRequest = (id: number | string) => ({
+    jsonrpc: jsonrpc2,
+    id,
+    method: 'foo',
+  });
+  const getMockResponse = () => ({ jsonrpc: jsonrpc2, id: 'foo' });
+
+  beforeEach(() => {
+    jest.useFakeTimers({ now: 10 });
+  });
+
+  it('forwards requests with ids seen for the first time', () => {
+    const filterFn = createDupeReqFilterMiddleware();
+    const nextMock = jest.fn();
+    const endMock = jest.fn();
+
+    filterFn(getMockRequest(1), getMockResponse(), nextMock, endMock);
+
+    expect(nextMock).toHaveBeenCalledTimes(1);
+    expect(endMock).not.toHaveBeenCalled();
+  });
+
+  it('ends the request if the id has been seen before', () => {
+    const filterFn = createDupeReqFilterMiddleware();
+    const nextMock = jest.fn();
+    const endMock = jest.fn();
+
+    filterFn(getMockRequest(1), getMockResponse(), nextMock, endMock);
+
+    expect(nextMock).toHaveBeenCalledTimes(1);
+    expect(endMock).not.toHaveBeenCalled();
+
+    const response = getMockResponse();
+    filterFn(getMockRequest(1), response, nextMock, endMock);
+    expect('result' in response).toBe(false);
```

Comment:
This seems unnecessary, where in this testing environment can the response ever have a `result` property?

Reply:
Well, it's testing the behavior of `filterFn`, and we're ensuring that it doesn't attach a `result` to the response in this test case.

Reply:
Right but `filterFn` in this test will never attach `result` to the response. Unless you want to write more fleshed out mock `next` and `end` functions.

Reply:
_Right now_ it won't attach a `result`, but with this test we are specifying that it should not do it in the future either.